src/store/movies/saga.tsx

```diff
@@ -0,0 +1,43 @@
+import { put, call, takeEvery, all, fork } from 'redux-saga/effects';
+import * as actionCreators from './actions';
+import * as actionTypes from './types';
+
+import { getMovies, getMovieItem } from '../../services/Movies.service';
+
+
+export function* onLoadMovies({searchParams, sortBy, filter}: actionTypes.LoadMoviesAction) {
+  try {
+    const { data } = yield call(getMovies, searchParams, sortBy, filter);
```

Comment:
рекомендую делать такую обёртку:

```suggestion
    const { data } = yield call(() => getMovies(searchParams, sortBy, filter));
```

так если ты захочешь в будущем сделать рефакторинг и изменить интерфейс функции `getMovies` тебе и тайпскрипту будет легче найти все места, где эта функция вызывается, и ты не пропустишь ни одного. я на этом уже не раз обжигался, и теперь никогда не передаю в `call` больше одного аргумента

Reply:
хотя такое решение возможно сломает тесты саги. но как я уже сказал, мы саги не тестировали ни на одном проекте